### PR TITLE
Use correct case on JSX components to avoid errors

### LINKED ISSUE #60677: [Bug]: wrong parameters name
### Quick summary

In a JSX svg a `fill-rule` property must be written as `fillRule`. See https://github.com/Automattic/wp-calypso/blob/trunk/client/blocks/login/two-factor-authentication/push-notification-illustration.jsx#L39-L58

### Steps to reproduce

1. Visit /log-in/push
2. Check the JS console

### What you expected to happen

1. The JS console to not print an error

### What actually happened

1. The JS console print an error

### Context

_No response_

### Browser

_No response_

### Simple/Atomic

_No response_

### Other notes

_No response_

### Reproducibility

Consistent

### Severity

_No response_

### Available workarounds?

_No response_

### Workaround details

_No response_

### PR DESCRIPTION
**Pull request by mpkelly**
#### Proposed Changes

When copying SVG elements into React sometimes we forget to convert kebab case attributes to camel case e.g. `fill-rule` should be converted to `fillRule`. Another common mistake is to use `class` instead of `className`. Even though these attributes get applied to the DOM correctly, React outputs an error in dev mode which pollutes the log and makes it harder to spot more serious errors. 

#### Testing Instructions

Should be enough to review the changes one by one. Functionally nothing has changed except the errors mentioned above are now gone. Before this change you would see a message in the log like this:

image

Fixes #60677

### CODE REVIEW

**Comment by mpkelly** [comment]
I am taking a look at these two CI checks that failed. 

**Update**: the lint checks have been silenced. The icons don't seem to be direct replacements, and I think it's best to update the whole 02-blocks package in one go. @creativecoder - what do you think about this?

I can't see anything obvious in my changes that would break the e2e tests, so maybe just a transient issue. I will monitor the next run. 

**Update two**: the same test still fails. I now have tests running locally but this test does not fail when I run it. Maybe Grant K has some input here - please ping when you get around to this PR. 

image

**Comment by creativecoder** [comment]
This is looking good. I very much appreciate not having these warnings about incorrect SVG property names cluttering up the console!

As discussed, let's move the o2 blocks changes to a separate PR for easier deployment.

Noting here from Slack that it appears we should remove the `a8c/todo` block all together, as it's no longer used, and the `a8c/task` block is preferred (p1656437267187649-slack-C9UMZ71QD).

> This PR modifies the release build for editing-toolkit

I haven't yet found where the ETK plugin might be importing some of the code that's modified here (with a quick search), will keep looking. Maybe this will reveal itself when the PR is split.

**Comment by creativecoder** [comment]
> E2E Tests (mobile) (Web app) — TeamCity build failed 

I also tried a rerun here for the e2e tests, and it's still failing. Let's try merging trunk into this branch to see if that brings in a fix. Otherwise we have some debugging to do!

**Comment by mpkelly** [comment]
@creativecoder I deleted the todo code. The checks all passed in the last build too. Maybe we can release this one?

File: test/e2e/.gitignore
Comment (worldomonation) [inline test/e2e/.gitignore:25]: Wouldn't this lead to some of the files under `test/e2e/image-uploads` being ignored?

Comment (mpkelly) [reply test/e2e/.gitignore:25]: Hi, no, this only excludes images directly at the root and not in sub-folders. 

image

Comment (worldomonation) [reply test/e2e/.gitignore:25]: Gotcha, thanks. I missed that it didn't have `**/` preceding it, oops.

**Review by worldomonation — APPROVED** [review]
Changes are okay from Quality Squad/KitKat side of things.

File: test/e2e/.gitignore
Comment (creativecoder) [reply test/e2e/.gitignore:25]: I'm assuming this `.gitignore` change fixes the issues with the e2e tests not passing? If so, please add a code comment and/or update the PR description so we document the reason for the change.

**Review by creativecoder — DISMISSED** [review]
Based on the most recent slack comment about checking for instances of the todo block still existing in past post content, it looks like we can't flat out remove the block without first verifying the impact. (p1656498367913499/1656437267.187649-slack-C9UMZ71QD).

Either way, let's break out any changes in the apps folder into additional PRs that are separate from changes in `client/`. The deploy process is different for apps, and each app has it's own deploy step, so will be easier to manage as separate PRs.

File: test/e2e/.gitignore
Comment (mpkelly) [reply test/e2e/.gitignore:25]: @creativecoder no, this doesn't fix any tests. The files that are output pollute the git commit diff unless you manually delete them each time.

**Comment by mpkelly** [comment]
@creativecoder I have reverted the changes to the `o2-blocks` package. This PR now only changes the `client` package. Aside from the `todo` block, the only other change in the `o2-blocks` package is for a single attribute in the `p2-autocomplete` block. I wonder if that justifies a PR. Either way, let's check if we can remove the `todo` block code.

**Comment by creativecoder** [comment]
> @creativecoder I have reverted the changes to the o2-blocks package. 

Thanks!

> This PR now only changes the client package. Aside from the todo block, the only other change in the o2-blocks package is for a single attribute in the p2-autocomplete block. I wonder if that justifies a PR.

It's really up to you, it can work either way. There's another deploy procedure for changes in the apps folder that requires wpcom commit access--so if you have changes there in this PR, please wait to merge until you can pair with someone to do that deploy immediately after merging.

**Review by creativecoder — COMMENTED** [review]
I haven't tested all the places there are changes in Calypso, but looking at the code, this looks fine.

I'm still confused why the e2e tests are failing.

**Comment by mpkelly** [comment]
@creativecoder, yes, the test that failed previously also failed again, for no apparent reason, after my last commit. 

I have just rerun the tests now, and they all passed. I already checked the failing test before and found no apparent link to my changes. I think it must be flaky. Are you willing to approve this PR so I can deploy it?

File: test/e2e/.gitignore
Comment (daledupreez) [reply test/e2e/.gitignore:25]: @mpkelly, if this is an issue specific to running the tests, can we move the changes into a dedicated PR so we can clearly call out why we're making this change and note it's impact? This change is currently buried within this technically unrelated PR, and it will be difficult for future developers to work out why this change occurred.

File: apps/o2-blocks/src/todo/editor.js
Comment (daledupreez) [inline apps/o2-blocks/src/todo/editor.js:3]: Do we need these linting changes for `apps/o2-blocks` in this PR? From a personal perspective, I think they are adding noise to the PR, but if we do keep the changes, then we need to document that explicitly in the PR description.

File: client/components/domains/use-my-domain/transfer-or-connect/option-content.jsx
Comment (daledupreez) [inline client/components/domains/use-my-domain/transfer-or-connect/option-content.jsx:36]: These `<element ...></end>` => `<element ... />` transforms aren't documented in the PR description -- can we either add relevant notes or remove the changes?

File: client/components/post-schedule/docs/example.jsx
Comment (daledupreez) [inline client/components/post-schedule/docs/example.jsx:172]: Could you please document why the `href="#"` attributes were removed? This feels like it might have unintended consequences, so please remove if it wasn't something you intended to include.

**Review by daledupreez — COMMENTED** [review]
@mpkelly, I had a look at the changes and noticed some areas where we either need to trim down the content of the PR or add some explicit docs. Could you make some decisions about those?
I will see if I can do some functional testing in the meantime.

File: packages/site-picker/src/index.tsx
Comment (daledupreez) [inline packages/site-picker/src/index.tsx:27]: This feels like a bug that's worth calling out due to the fact that it's changing the accessibility of the logo.

That may be worth pushing into a dedicated PR, as it may also be worth simplifying this code slightly as follows:
```
if ( ! logo?.url ) {
  return globe;
}

return <img width="34" alt="" aria-hidden="true" src={ logo.url } />;
```

**Comment by mpkelly** [comment]
Based on the review comments, I think I will redo a Calypso-only PR to start with and then do some of the others in a separate PR.

**Comment by mpkelly** [comment]
I am closing this in favour of #65442, and a couple of other more focused PRs as per the feedback above.

**Pull request closed without merging**
